safekeeper/src/timeline.rs

```diff
@@ -339,10 +339,12 @@ impl Timeline {
         ttid: TenantTimelineId,
         wal_backup_launcher_tx: Sender<TenantTimelineId>,
         server_info: ServerInfo,
+        commit_lsn: Lsn,
+        local_start_lsn: Lsn,
     ) -> Result<Timeline> {
```

Comment:
Since we now can accept arbitrary `commit_lsn` and `local_start_lsn`, it's better to check validity (`commit_lsn >= local_start_lsn`) of them, I'd do that in `SharedState::create_new`.